### PR TITLE
[SCV-34][SCV-29][SCV-18] Move Metadata to Assets and Update Metadata Types

### PR DESCRIPTION
**Pull request by quetcodesfire**
Here I moved metadata to assets and updated `createLink` to create types based on file extension names.

### CODE REVIEW

File: search/lib/convert/granules.js [hunk 1]
```diff
@@ -117,8 +119,8 @@ function cmrGranToFeatureGeoJSON (event, cmrGran) {
       {
         rel: 'root',
         href: generateAppUrl(event)
-      },
-      wfs.createLink('metadata', cmr.makeCmrSearchUrl(`/concepts/${cmrGran.id}.native`))
+      }
+      // wfs.createLink('metadata', cmr.makeCmrSearchUrl(`/concepts/${cmrGran.id}.native`))
```
Comment (DrewE84) [inline search/lib/convert/granules.js:123]: before we merge to master, we should remove these commented-out lines.

File: search/tests/convert/granules.spec.js [hunk 2]
```diff
@@ -244,14 +246,22 @@ describe('granuleToItem', () => {
...
             provider: 'USA',
             datetime: '0',
             start_datetime: '0',
             end_datetime: '1'
           },
           type: 'Feature',
-          assets: {},
+          assets: {
+            metadata: {
+              href: 'https://cmr.earthdata.nasa.gov/search/concepts/1.native',
```
Comment (DrewE84) [inline search/tests/convert/granules.spec.js:259]: Just looking for clarification- is line 23 of `lib/util/app.js` supposed to transform this resulting href into a .xml file?

File: search/lib/util/app.js [hunk 3]
```diff
@@ -11,9 +11,27 @@ const extractParam = (queryStringParams, param, defaultVal = null) => {
   return defaultVal;
 };
 
-const createLink = (rel, href, title, type = 'application/json') => ({
-  href, rel, type, title
-});
+// function createLink(rel, href, title, type = 'application/json') {
+
+// }
```
Comment (DrewE84) [inline search/lib/util/app.js:16]: found another commented out line to remove.

File: search/lib/convert/granules.js (hunk 1 above)
Comment (quetcodesfire) [reply search/lib/convert/granules.js:123]: Good catch!

File: search/tests/convert/granules.spec.js (hunk 2 above)
Comment (quetcodesfire) [reply search/tests/convert/granules.spec.js:259]: Yup, didn't notice that didn't convert. I've updated the code here.

File: search/lib/util/app.js (hunk 3 above)
Comment (quetcodesfire) [reply search/lib/util/app.js:16]: Deleted.

**Comment by DrewE84** [comment]
Looks good to me. It seems to have a few merge conflicts, but once those are resolved it should be good to go.

**Pull request merged**